src/services/analysis.ts

```diff
@@ -0,0 +1,50 @@
+import axios from 'axios'
+
+// Utils
+import { delay, getRequestHeaders } from '../utils'
+
+export const analyze = async (data: {}, accessToken: string, url: string) => {
+  const headers = getRequestHeaders(accessToken)
+
+  try {
+    const response = await axios.post(url, { data }, { headers })
+
+    const uuidUrl = `${url}/${response.data.uuid}`
+    let status = response.data.status
+
+    // Initial status check
+    if (status === 'Finished') {
+      return getIssues(accessToken, `${uuidUrl}/issues`)
+    }
+
+    for (let i = 0; i < 10; i++) {
+      await delay(3000)
```

Comment:
Can we just have a setTimeout in here? It seems a bit too much having a function that just does something that a native method does already :)

Reply:
It's better to have a Promise wrapped method for setTimeout to make the best use of `async/await` across the codebase